quinn-proto/src/assembler.rs

```diff
@@ -86,12 +86,17 @@ impl Assembler {
         let new = BinaryHeap::with_capacity(self.data.len());
         let old = mem::replace(&mut self.data, new);
         for chunk in old.into_sorted_vec().into_iter().rev() {
-            if offset + (buffer.len() as u64) < chunk.offset {
+            let end = offset + (buffer.len() as u64);
+            let overlap = end.saturating_sub(chunk.offset);
```

Comment:
We can defer the initial definition of `overlap` until after the `if end < chunk.offset` and inline the cast to `usize`. Let's also introduce an empty line before that definition.

Reply:
I think this whole thing could actually be recast into `if let Some(overlap) = end.saturating_sub(chunk.offset) { /* append */ } else { /* split */ }`, if we want to get really tidy.

Reply:
That sounds even better!

Reply:
I just updated to `checked_sub`. I think it's possibly not quite what you were aiming for as the `extend_from_slice` has to happen depending on a different condition: `overlap < chunk.bytes.len()`. I also updated that condition to avoid the need to redefine `overlap`. Let me know what you think?